tietokanta/src/main/resources/db/migration/R__Laskutusyhteenveto_mhu.sql

```diff
@@ -671,11 +684,66 @@ BEGIN
+            akilliset_laskutetaan := 0.0;
+            vahingot_laskutettu := 0.0;
+            vahingot_laskutetaan := 0.0;
+
+            FOR akilliset_rivi IN SELECT summa AS akilliset_summa, l.erapaiva AS erapaiva, lk.maksueratyyppi
+                                     FROM kulu l
+                                              JOIN kulu_kohdistus lk ON lk.kulu = l.id
+                                              JOIN toimenpideinstanssi tpi
+                                                   ON lk.toimenpideinstanssi = tpi.id AND tpi.id = t.tpi
+                                     WHERE lk.maksueratyyppi IN ('akillinen-hoitotyo', 'muu')
+                                       AND lk.poistettu IS NOT TRUE
+                                       AND l.erapaiva BETWEEN hk_alkupvm AND aikavali_loppupvm
+                LOOP
+	                
+	                -- Äkilliset hoitotyöt
+	                IF akilliset_rivi.maksueratyyppi = 'akillinen-hoitotyo' THEN
+	                
+	                    SELECT akilliset_rivi.akilliset_summa AS summa, 0::NUMERIC AS korotus INTO akilliset;
+	                    RAISE NOTICE 'akilliset_rivi - hoitotyöt: %', akilliset;
+	                   
+	                    IF akilliset_rivi.erapaiva <= aikavali_loppupvm THEN
+	                        -- Hoitokauden alusta
+	                        akilliset_laskutettu := akilliset_laskutettu + COALESCE(akilliset.summa, 0.0);
+	
+	                        IF akilliset_rivi.erapaiva >= aikavali_alkupvm AND
+	                           akilliset_rivi.erapaiva <= aikavali_loppupvm THEN
+	                            -- Laskutetaan nyt
+	                            akilliset_laskutetaan := akilliset_laskutetaan + COALESCE(akilliset.summa, 0.0);
+	                        END IF;
+	                    END IF;
+	                    RAISE NOTICE 'Äkilliset: %', akilliset;
+                   END if;
+                   
+                   -- Vahingot
+                   IF akilliset_rivi.maksueratyyppi = 'muu' THEN
+                   
+	                    SELECT akilliset_rivi.akilliset_summa AS summa, 0::NUMERIC AS korotus INTO akilliset;
+	                    RAISE NOTICE 'akilliset_rivi - vahingot: %', akilliset;
+	                   
+	                    IF akilliset_rivi.erapaiva <= aikavali_loppupvm THEN
+	                        -- Hoitokauden alusta
+	                        vahingot_laskutettu := vahingot_laskutettu + COALESCE(akilliset.summa, 0.0);
+	
+	                        IF akilliset_rivi.erapaiva >= aikavali_alkupvm AND
+	                           akilliset_rivi.erapaiva <= aikavali_loppupvm THEN
+	                            -- Laskutetaan nyt
+	                            vahingot_laskutetaan := vahingot_laskutetaan + COALESCE(akilliset.summa, 0.0);
+	                        END IF;
+	                    END IF;
+	                    RAISE NOTICE 'Vahingot: %', akilliset;
+                   END IF;
+                END LOOP;
```

Comment:
Rivillä 756 haetaan hankintojen summat. Siellä tällä hetkellä filtteröidään ulos vain lisätyöt. Eikö tämä tarkoita sitä, että nyt hankintoihin tulee sekä äkilliset että vahingonkorvaukset kahdesti, kun niitä ei erikseen irroteta rivillä 756 hankinnoista?

Reply:
Kyllä, olet oikeassa, ja tästä olikin palaverissa puhetta.
Lisäksi yhteensä-sarakkeeseen ei laskettu äkillisiä hoitotöitä & vahinkoja.
Korjattu [a96550a](https://github.com/finnishtransportagency/harja/pull/2800/commits/a96550aec2c2ab07b31f0173dd4611f8ff18d308)